src/Microsoft.ManagedZLib/src/TreeDesc.cs

```diff
@@ -0,0 +1,310 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.ManagedZLib;
+
+public class TreeDesc
+{
+    public CtData[] _dynamicTree;
+    public int _maxCode;
+    public StaticTreesDesc _StaticTreeDesc;
+
+    public TreeDesc(CtData[] dynamicTree, StaticTreesDesc StaticTreeDesc)
+    {
+        _dynamicTree = dynamicTree;
+        _StaticTreeDesc = StaticTreeDesc;
+        _maxCode = 0;
+    }
+
+    // Reverse the first len bits of a code, using straightforward code
+    // (a faster method would use a table)
+    // Code: the value to invert
+    // Len: its bit length 
+    public static uint BitReverse(uint code, int Len)
+    {
+        //int Len = Marshal.SizeOf(code) * 8;
+        Debug.Assert((1 <= Len) && (Len <= 15));
+        uint res = 0;
+        do
+        {
+            res |= code & 1;
+            code >>= 1;
+            res <<= 1;
+        } while (--Len > 0);
+        return res >> 1;
+    }
+    //Generate the codes for a given tree and bit counts (which need not be
+    //optimal).
+    //tree - the tree to decorate 
+    //_maxCode - largest code with non zero frequency
+    //codeCount - number of codes at each bit length 
+    public void GenCodes(Span<ushort> codeCount)
+    {
+        ushort[] nextCode = new ushort[DeflateTrees.MaxBits + 1]; /* next code value for each bit length */
+        uint code = 0;         /* running code value */
+        int bits;                  /* bit index */
+        int n;                     /* code index */
+
+        // The distribution counts are first used to generate
+        // the code values without bit reversal.
+        for (bits = 1; bits <= DeflateTrees.MaxBits; bits++)
+        {
+            code = (code + codeCount[bits - 1]) << 1;
+            nextCode[bits] = (ushort)code;
+        }
+
+        // Check that the bit counts in code's count (codeCount) are consistent.
+        // The last code must be all ones.
+        Debug.Assert(code + codeCount[DeflateTrees.MaxBits] - 1 == (1 << DeflateTrees.MaxBits) - 1,
+                "inconsistent bit counts");
+
+        for (n = 0; n <= _maxCode; n++)
+        {
+            int len = _dynamicTree[n].Len;
+            if (len == 0) continue;
+            /* Now reverse the bits */
+            _dynamicTree[n].Freq = (ushort)BitReverse(nextCode[len]++, len);
+        }
+    }
+
+    // Index within the heap array of least frequent node in the Huffman tree
+    public const int Smallest = 1;
+
+    // Remove the smallest element from the heap and recreate the heap with
+    // one less element.Updates heap and heap_len.
+    public void PriorityQueueRemove(out int removedTop, int[] Heap, ref int heapLen, byte[] depth)
+    {
+        removedTop = Heap[Smallest];
+        Heap[Smallest] = Heap[heapLen--];
+        PriorityQueueDownHeap(Smallest, Heap, ref heapLen, depth);
+    }
```

Comment:
Argument names should be lowercase:
```suggestion
    public void PriorityQueueRemove(out int removedTop, int[] heap, ref int heapLen, byte[] depth)
    {
        removedTop = heap[Smallest];
        heap[Smallest] = Heap[heapLen--];
        PriorityQueueDownHeap(Smallest, heap, ref heapLen, depth);
    }
```

Same in the rest of the methods.